.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,65 @@
+name: Run CodeQL
+
+on:
+  push:
+    branches: [ master ]
+    paths-ignore:
+      - '**/*.md'
+  pull_request:
+    branches: [ master ]
+    paths-ignore:
+      - '**/*.md'
+  schedule:
+    - cron: '24 2 * * 4'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'csharp' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python' ]
+        # Learn more:
+        # https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#changing-the-languages-that-are-analyzed
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup .NET Core
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.x
```

Comment:
@Bond-009 do we do anything similar in other C# repos? Perhaps we should be more or less strict with the version, feel free to weigh in here.

Reply:
We use an exact version in the server repo, but this is fine for me

Reply:
It is no GitHub CI workflow, but this is how u did it in the JF main repo on Azure CI:

```yaml
# ...
      - task: UseDotNet@2
        displayName: "Install .NET SDK 5.x"
        condition: eq(variables['ImageName'], 'ubuntu-latest')
        inputs:
          packageType: sdk
          version: '5.x'
# ...
```

Reply:
I am blind and did not see that :facepalm: (it is `5.0.100` in the main repo as far as I can tell)

Reply:
I think using `5.0.x` is fine here so we don't have to keep updating it for patch versions.
Last I checked the workflow doesn't support `5.x`

Reply:
> I think using `5.0.x` is fine here so we don't have to keep updating it for patch versions.
> Last I checked the workflow doesn't support `5.x`

Just tested that, since I was not sure either, nope `5.x` will unfortunately fail.